### PR TITLE
WString and Misc changes

### PR DESCRIPTION
**Pull request by gabryelreyes**
- Added std::string-based WString
- Fixed Arduino.h line endings from LF to CRLF
- Updated CI workflows to latest versions
- Added CI tests
- Bumped version to 0.2.0

### CODE REVIEW

File: .github/workflows/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+platformio==6.1.15
```
Comment (BlueAndi) [inline .github/workflows/requirements.txt]: Why is this necessary? Couldn't we go always with the latest here?

File: src/WString.h [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ * @brief  String implementation for test
+ * @author Andreas Merkle <web@blue-andi.de>
+ * @author Luca Dubies <luca.dubies@newtec.de>
+ * @author Gabryel Reyes <gabryelrdiaz@gmail.com>
+ *
```
Comment (BlueAndi) [inline src/WString.h:32]: Isn't that a complete take over from Pixelix?

**Review by BlueAndi — CHANGES_REQUESTED** [review]

File: src/WString.h (hunk 2 above)
Comment (gabryelreyes) [reply src/WString.h:32]: No, I had to add some methods. Concat and replace

File: .github/workflows/requirements.txt (hunk 1 above)
Comment (gabryelreyes) [reply .github/workflows/requirements.txt]: This is the latest version available via PIP

Comment (gabryelreyes) [reply .github/workflows/requirements.txt]: And the requirements.txt file is required for caching using the GitHub Action

Comment (BlueAndi) [reply .github/workflows/requirements.txt]: Looks different than recommended: https://docs.platformio.org/en/stable/integration/ci/github-actions.html

File: src/WString.h (hunk 2 above)
Comment (BlueAndi) [reply src/WString.h:32]: Ah yes, I forgot. :-)

File: .github/workflows/requirements.txt (hunk 1 above)
Comment (gabryelreyes) [reply .github/workflows/requirements.txt]: It shows the default variant, and I believe it is outdated as it uses setup-python@v4. 
This new caching uses is integrated in the setup-python@v5
